go/common/servenv/servenv.go

```diff
@@ -166,25 +167,14 @@ func (se *ServEnv) SetListeningURL(u url.URL) {
 	se.listeningURL = u
 }
 
-// PopulateListeningURL sets the listening URL based on hostname and port
+// PopulateListeningURL sets the listening URL based on the configured hostname and port.
+// The hostname should already be set by Init() before this is called.
 func (se *ServEnv) PopulateListeningURL(port int32) {
-	host, err := netutil.FullyQualifiedHostname()
-	if err != nil {
-		slog.Warn("Failed to get fully qualified hostname, falling back to simple hostname",
-			"error", err,
-			"note", "This may indicate DNS configuration issues but service will continue normally")
-		host, err = os.Hostname()
-		if err != nil {
-			slog.Error("os.Hostname() failed", "err", err)
-			os.Exit(1)
```

Comment:
os.Hostname() was only used here for logging, and it seemed excessive to kill the whole process because a log-only call fails. Code's changed this to log the same hostname that's actually used as the listening URL and the os.Exit() call is removed entirely